src/Stratis.Bitcoin.IntegrationTests/SmartContracts/SmartContractWalletOnPosNetworkTests.cs

```diff
@@ -0,0 +1,194 @@
+                ContractStateRepositoryRoot senderState = scSender.FullNode.NodeService<ContractStateRepositoryRoot>();
+                ContractStateRepositoryRoot receiverState = scReceiver.FullNode.NodeService<ContractStateRepositoryRoot>();
+                uint160 tokenContractAddress = transferContractTransaction.GetNewContractAddress();
+                Assert.NotNull(senderState.GetCode(tokenContractAddress));
+                Assert.NotNull(receiverState.GetCode(tokenContractAddress));
+                scSender.FullNode.MempoolManager().Clear();
+
+                // Create a transfer token contract
+                compilationResult = SmartContractCompiler.CompileFile("SmartContracts/TransferTestPos.cs");
+                Assert.True(compilationResult.Success);
+                contractCarrier = SmartContractCarrier.CreateContract(vmVersion, compilationResult.Compilation, gasPrice, gasLimit);
+                contractCreateScript = new Script(contractCarrier.Serialize());
+                txBuildContext = new TransactionBuildContext(scSender.FullNode.Network, new WalletAccountReference(WalletName, AccountName), new[] { new Recipient { Amount = 0, ScriptPubKey = contractCreateScript } }.ToList(), Password)
+                {
+                    ChangeAddress = senderAddress,
+                    MinConfirmations = maturity,
+                    FeeType = FeeType.High
+                };
+
+                // Build the transfer contract transaction
+                transferContractTransaction = BuildTransferContractTransaction(scSender, txBuildContext);
+
+                // Add the smart contract transaction to the mempool to be mined.
+                scSender.AddToStratisMempool(transferContractTransaction);
+
+                // Wait for the token transaction to be picked up by the mempool
+                TestHelper.WaitLoop(() => scSender.CreateRPCClient().GetRawMempool().Length > 0);
+                scSender.GenerateStratisWithMiner(1);
+
+                // Ensure the node is synced
+                TestHelper.WaitLoop(() => TestHelper.IsNodeSynced(scSender));
+
+                // Ensure both nodes are synced with each other
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(scReceiver, scSender));
+
+                // Ensure that boths nodes has the contract
+                senderState = scSender.FullNode.NodeService<ContractStateRepositoryRoot>();
+                receiverState = scReceiver.FullNode.NodeService<ContractStateRepositoryRoot>();
+                tokenContractAddress = transferContractTransaction.GetNewContractAddress();
+                Assert.NotNull(senderState.GetCode(tokenContractAddress));
+                Assert.NotNull(receiverState.GetCode(tokenContractAddress));
+                scSender.FullNode.MempoolManager().Clear();
+
+                //THE TESTS FAIL ON TRANSFER CALLS
+
+                // Create a call contract transaction which will transfer funds
```

Comment:
delete or fix the section commented here

Reply:
@majikandy this test is now fixed 👍